libs/remix-ui/settings/src/lib/remix-ui-settings.tsx

```diff
@@ -0,0 +1,171 @@
+            <a href="https://medium.com/p/66ef69e14931/" target="_blank"> Analytics in Remix IDE</a> <span>&</span> <a target="_blank" href="https://matomo.org/free-software">Matomo</a>
+          </label>
+        </div>
+      </div>
+    </div>
+  )
+
+  const saveToken = () => {
+    setSaveTokenState(true)
+    props.config.set('settings/gist-access-token', tokenValue)
+  }
+
+  const removeToken = () => {
+    setRemoveTokenState(true)
+    setTokenValue('')
+    props.config.set('settings/gist-access-token', '')
+  }
+
+  const handleSaveTokenState = useCallback(
+    (event) => {
+      setTokenValue(event.target.value)
+    },
+    [tokenValue]
+  )
+
+  const gistToken = () => (
+    <div className="border-top">
+      <div className="card-body pt-3 pb-2">
+        <h6 className="card-title">{ gitAccessTokenTitle }</h6>
+        <p className="mb-1">{ gitAccessTokenText }</p>
+        <p className="">{ gitAccessTokenText2 }</p>
+        <p className="mb-1"><a className="text-primary" target="_blank" href="https://github.com/settings/tokens">{ gitAccessTokenLink }</a></p>
+        <div className=""><label>TOKEN:</label>
+          <div className="text-secondary mb-0 h6">
+            <input id="gistaccesstoken" data-id="settingsTabGistAccessToken" type="password" className="form-control" onChange={handleSaveTokenState} value={ tokenValue } />
+            <div className="d-flex justify-content-end pt-2">
+              <CopyToClipboard content={tokenValue} data-id='copyToClipboardCopyIcon' />
+              <input className="btn btn-sm btn-primary ml-2" id="savegisttoken" data-id="settingsTabSaveGistToken" onClick={() => saveToken()} value="Save" type="button" disabled={tokenValue === ''}></input>
+              <button className="btn btn-sm btn-secondary ml-2" id="removegisttoken" data-id="settingsTabRemoveGistToken" title="Delete Github access token" onClick={removeToken}>Remove</button>
+            </div>
+          </div></div>
+      </div>
+    </div>
+  )
+
+  const themes = () => {
+    const themes = props._deps.themeModule.getThemes()
+    if (themes) {
+      return themes.map((aTheme, index) => (
+        <div className="radio custom-control custom-radio mb-1 form-check" key={index}>
+          <input type="radio" onChange={event => { onswitchTheme(event, aTheme.name) }} className="align-middle custom-control-input" name='theme' id={aTheme.name} data-id={`settingsTabTheme${aTheme.name}`} checked = {props._deps.themeModule.active === aTheme.name ? true : null}/>
+          <label className="form-check-label custom-control-label" data-id={`settingsTabThemeLabel${aTheme.name}`} htmlFor={aTheme.name}>{aTheme.name} ({aTheme.quality})</label>
+        </div>
+      )
+      )
+    }
+  }
+
+  return (
+    <div>
+      {saveTokenState ? <Toaster message='Access token has been saved' /> : null}
```

Comment:
We can find a way to manage toasters in an array.

Reply:
refactored `Toaster` into its own actions and reducer while removing the code duplicate from line `157` and `158`